cranelift/codegen/src/isle_prelude.rs

```diff
@@ -935,6 +935,10 @@ macro_rules! isle_common_prelude_methods {
             val as u16
         }
 
+        fn u64_as_u16_extractor(&mut self, val: u64) -> Option<u16> {
+            Some(val as u16)
+        }
```

Comment:
Personally I'm always wary to discard the lower bits with `as` in lowering rules lest they accidentally get removed. Given that `iconst` extraction is generally signed could the above rules do `(i16_from_iconst imm)` and then the immediate to the instruction is `(i16_as_u16 imm)` to be clear that there's no data loss it's just reinterpreting the sign bit?